src/coreComponents/mesh/generators/NewGhosting.cpp

```diff
@@ -654,45 +769,68 @@ std::tuple< MeshGraph, GhostRecv, GhostSend > performGhosting( MeshGraph const &
   std::vector< std::vector< TriGlbIdx > > const & indices = adjacency.indices;
   std::vector< std::vector< TriScalarInt > > const  & values = adjacency.values;
 
+  // Makes a Trilinos tools MPI communicator and return smart pointer to it
   RCP< TriComm const > const comm = make_rcp< Teuchos::MpiComm< int > const >( MPI_COMM_GEOSX );
+  // Makes a const TriMap, passing the arguments to the constructor of TriMap, returns smart pointer to it 
+  // note TriMap = Tpetra::Map< TriLocIdx, TriGlbIdx >, created using this constructor https://docs.trilinos.org/dev/packages/tpetra/doc/html/classTpetra_1_1Map.html#a4a21c6654fc9fb6db05644a52cc0128d
+  // Ultimately just collecting who owns what parts of the global matrix
   auto const ownedMap = make_rcp< TriMap const >( Tpetra::global_size_t( n ), ownedGlbIdcs.data(), TriLocIdx( numOwned ), TriGlbIdx{ 0 }, comm );
 
   // The `upward` matrix offers a representation of the graph connections.
   // The matrix is square. Each size being the number of geometrical entities.
+  // The matrix is just the assembed, global adjacency.
+  // Note that the matrix will end up being lower triangular, and banded because edge rows only have nonzeros in the node columns, faces only have nonzeros with edges, etc
```

Comment:
In fact, it's lower triangular for "std" meshes. This may become different with fractures and cell to cell dependencies _e.g._.